compiler/luci/partition/include/luci/Partition.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_PARTITION_H__
+#define __LUCI_PARTITION_H__
+
+#include <luci/IR/Module.h>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace luci
+{
+
+/**
+ * @brief PartitionTable holds partition information
+ */
+struct PartitionTable
+{
+  std::vector<std::string> groups;
+  std::string def_group;
+
+  // assign by opcode: OPCODE=group
+  std::unordered_map<std::string /* OPCODE */, std::string /* group */> byopcodes;
+
+  // TODO add assign by OP name
+};
+
+/**
+ * @brief PartedModule holds partitioned module and group name
+ */
+struct PartedModule
+{
+  std::unique_ptr<Module> module;
+  // group name used to partition this module
+  std::string group;
+
+  // unique name(filename) of this module
+  std::string name;
+};
+
```

Comment:
`PartedModule` is a bit confusing to me. 

Here's what I thought. There is a similar relationship, e.g. `Module` and `Graph`. And `Module` has a vector of graph. And `Graph` class is from `SubGraph` in schema and `Module` is from `Model`.
```flatbuffers
// circle_schema.fbs
table SubGraph  { ... }
table Model {
  subgraphs:[SubGraph];
}
```

 So, I first thought `PartedModule` would be like this.

```cxx
struct PartedModule
{
  std::vector<std::unique_ptr<PGraphs>> PGraphs; // PGraphss(?) ``)
  // group name used to partition this module
  std::string group;

  // unique name(filename) of this module
  std::string name;
};
```

Or, it assumes there's more than one circle model?

Reply:
`PartedModule` has no direct relation with `PGraph` or `PGraphs`.
One `Module` is from/to one circle file.
`PartedModule` is a wrapper for a `Module` as a result of partitioning.
After partitioning, one `Module` will become one or more `PartedModule`, that is `PartedModules`, vector of `PartedModule`

Reply:
`PGraph` and `PGraphs` are internal structure used while partitioning.

Reply:
@seanshpark Oh, I thought `Module` would split into `PGraphs`. 

Then, `Module` will split into `PartedModule`s after partitioning and `PartedModule` consists of `PGraphs` internally. Am I right?

Reply:
> Then, Module will split into PartedModules after partitioning 

soft of like this.

> and PartedModule consists of PGraphs internally. Am I right?

Nope.  It just has a `Module` and some strings for additional information. `Module` is existing `luci::Module`
```cxx
struct PartedModule
{
  std::unique_ptr<Module> module;
  // group name used to partition this module
  std::string group;

  // unique name(filename) of this module
  std::string name;
};
```

Reply:
When not considering partition,

`luci::Module` = `loco::Graph` + `loco::Graph` + ...
`loco::Graph` = `luci::CircleNode` + `luci::CircleNode` + ...

I think it is clear.

---

When considering partition,

`luci::Module` --> `PartedModule` + `PartedModule` + ... = `PartedModules`
`PartedModule` = `luci::Module` + group(I don't know what is this exactly) + unique file name 

Is it right...?

Reply:
> Is it right...?

Seems right :)

> group(I don't know what is this exactly)

`group` is a key to use how to partition.
partitioning is like gather the nodes with same `group` and put it in a `PartedModule`... 
with some conditions, one node should be connected to another and so on...

Reply:
> `luci::Module` --> `PartedModule` + `PartedModule` + ... = `PartedModules`
> `PartedModule` = `luci::Module` + group(I don't know what is this exactly) + unique file name 

It looked like following equation for me...

- `luci::Module total` = `PartedModuleA` + `PartedModuleB` + `PartedModuleC`
- `PartedModuleA` = `luci::Module total` + group + name =  `PartedModuleA` + `PartedModuleB` + `PartedModuleC` + group + name
- Then, `PartedModuleB` + `PartedModuleC` = 0?

What is wrong with above flow...?

Reply:
And one more question with it...! (If this is not proper question in this PR, I will move this comment to somewhere @.@)

You said that `PGraph` and `PartedModule` is not related with each other.
But what makes me confused is the relation between `luci::Module` and `loco::Graph`.
(`luci::Module` = `loco::Graph` + `loco::Graph` + ...)
With this convention, I thought that `PartedModule` = `PGraph` + `PGraph` + ... at first.
When `PGraph` is used for...?

Reply:
> PartedModule = luci::Module + ...

` group + name` is some helper string so we can remove this from Eq.

It is like

`luci::Module` = part A of  `luci::Module` + part B of `luci::Module` + part C of `luci::Module`

part A of `luci::Module` = `PartedModuleA`
part B of `luci::Module` = `PartedModuleB`
part C of `luci::Module` = `PartedModuleC`

`luci::Module` = `PartedModuleA` + `PartedModuleB` + `PartedModuleC`